src/CMakeLists.txt

```
@@ -0,0 +1,36 @@
+# libcps
+add_library(libcps cps/env.cpp cps/loader.cpp cps/printer.cpp cps/search.cpp
+                   cps/utils.cpp cps/version.cpp)
```

```suggestion
add_library(libcps
  cps/env.cpp
  cps/loader.cpp
  cps/printer.cpp
  cps/search.cpp
  cps/utils.cpp
  cps/version.cpp
  )
```
Not a big deal, but putting each file on a different source line _generally_ makes it easier to trivially rebase multiple PRs that add or remove files. Trivial rebasing is really nice when trying to get PRs to land from the GitHub web UI.

---

Also, I'd just call the library `cps` and not `libcps`, but I don't think I can make a simple PR suggestion to clean that up.

---

Meson will create a library with the common scheme for each platform. On Unices this will be `libcps.[a,so,dylb]`, on windows it'll be `cps.[a,lib,dll]`. I'm not sure what CMake does here.

---

If you name the library `cps`, CMake will make a `libcps.a` or `libcps.so`, depending on whether you selected a static or shared build. The default is static, but someone building a project can choose shared with `-DBUILD_SHARED_LIB=ON`.

The big thing is to ship a library named `libcps.a`, yes. We could add a little validation to the CI if that's important to get right. Eventually, if we add support for Conan, there are ways to add a "packaging test" that includes linking against packaged libraries, which will be a more end-to-end way to ensure the same effect.